nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -131,9 +135,19 @@
     @WritesAttribute(attribute = "invokehttp.java.exception.message", description = "The Java exception message raised when the processor fails"),
     @WritesAttribute(attribute = "user-defined", description = "If the 'Put Response Body In Attribute' property is set then whatever it is set to "
         + "will become the attribute key and the value would be the body of the HTTP response.")})
-@DynamicProperty(name = "Header Name", value = "Attribute Expression Language", expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
-                    description = "Send request header with a key matching the Dynamic Property Key and a value created by evaluating "
-                            + "the Attribute Expression Language set in the value of the Dynamic Property.")
+@DynamicProperties ({
+    @DynamicProperty(name = "Header Name", value = "Attribute Expression Language", expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
+        description =
+            "Send request header with a key matching the Dynamic Property Key and a value created by evaluating "
+                + "the Attribute Expression Language set in the value of the Dynamic Property."),
+    @DynamicProperty(name = "post.form.<NAME>", value = "Attribute Expression Language", expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
+        description =
+            "When the HTTP Method is POST, dynamic properties with the property name in the form of post.form.<NAME>,"
+                + " where the <NAME> will be the form data name, will be used to fill out the multipart form parts."
+                + "  If the value is the literal 'FLOWFILE_CONTENT', that form part will have the flowfile's data."
```

Comment:
What do you think about instead adding a new, optional property along the lines of "Content Multipart/Form Name", along with maybe another property that indicates whether or not use Multipart/Form encoding?
Going that route, I think, may make the configuration a little bit annoying. But the reason that I suggest it is two-fold:
1) As-is, this PR changes the behavior of the user-defined properties. If someone already was using a post.form.XYZ property we'd change behavior. This is only a slight concern given that it would be a pretty weird HTTP Header to send :)
2) It's difficult to remember the exact string contents "FLOWFILE_CONTENT" and if there's ever a typo, the consequences could be pretty bad. The user will think everything was properly uploaded and have no idea that they are not actually sending the FlowFile's content but are sending some string such as FLOWFIEL_CONTENT every time.

Reply:
I agree that this is more awkward than I would like, but that is a general problem with the was that properties work.

A form can have an arbitrary number of fields or parts.  It also may not be for uploading a document.  So to make it broadly useful you have to be able to define multiple fields.  Given the way that flowfiles work, I think only being able to define 1 'file upload' field for the flow file is good enough.

I thought that using this technique, like I did for the ExecuteStreamCommand to support arguments and fix the escaping issue, would allow for having _N_ parts, with EL support available.  Now, new properties for the things that I have as known constant values _is_ a better way to go, but I was hesitant to add new properties.

I think the most flexible solution is to use the dynamic parameters, with new parameters for the content  Content form name, Content file name.

Reply:
Also, we could change the name pattern to be "nifi.post.form"

Reply:
Yeah I think we're on the same page here. The first point above is a bit of a minor quibble that I'm not really concerned about. It's "breaking" but.... would be a super specific corner case that I don't believe anyone would ever encounter. If they do, and they really had to have that, it could be addressed by dropping an old NAR in or whatever.

But the second point, about requiring the magical value of FLOWFILE_CONTENT is a much more realistic concern. Using a well-known property name for that, I think, is a good idea. It's okay to add new properties as but we want to avoid adding properties that either (a) change behavior (i.e., default should be backward compatible) or (b) make the processor invalid (i.e., don't add a new required property without a default value)

Reply:
Regarding other comment, I think I would just go with "post.form" and not "nifi.post.form". It's a very unlikely corner case so I'd prefer not to add the overhead of typing "nifi." at the beginning every time.

Reply:
Or, if you want to be even more clever about not breaking backward compatibility you could use the notation "post:form:<name>" because other user-defined properties are treated as headers, and colons are not legal in header names (https://www.w3.org/Protocols/rfc2616/rfc2616-sec2.html#sec2 specifies what a 'token' is and https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html specifies that a header name is simply a 'token').

Reply:
This is what i did